From 6b0a4665c760d25362752cc57ee239bb19e4f87d Mon Sep 17 00:00:00 2001
From: Justin Waters <justin.waters@timesys.com>
Date: Wed, 9 Mar 2016 13:16:39 -0500
Subject: [PATCH 3/7] ge_bx50v3: Add support for FSL Community Yocto images

The Freescale community images use two partitions: one for
boot files, the other for the RFS. This option makes the default
configurable at build time.

Signed-off-by: Justin Waters <justin.waters@timesys.com>
---
 board/ge/bx50v3/Kconfig     |  4 ++++
 include/configs/ge_bx50v3.h | 36 ++++++++++++++++++++++++------------
 2 files changed, 28 insertions(+), 12 deletions(-)

diff --git a/board/ge/bx50v3/Kconfig b/board/ge/bx50v3/Kconfig
index d50dece..320b3b8 100644
--- a/board/ge/bx50v3/Kconfig
+++ b/board/ge/bx50v3/Kconfig
@@ -15,4 +15,8 @@ config SYS_SOC
 config SYS_CONFIG_NAME
 	default "ge_bx50v3"
 
+config YOCTO_IMAGE
+	bool "Use yocto image layout"
+        default n
+
 endif
diff --git a/include/configs/ge_bx50v3.h b/include/configs/ge_bx50v3.h
index 6fa4a9a..4b3c25e 100644
--- a/include/configs/ge_bx50v3.h
+++ b/include/configs/ge_bx50v3.h
@@ -17,21 +17,31 @@
 
 #if defined(CONFIG_TARGET_GE_B450V3)
 #define CONFIG_BOARD_NAME	"General Electric B450v3"
-#define CONFIG_DEFAULT_FDT_FILE	"/boot/imx6q-b450v3.dtb"
+#define CONFIG_DEFAULT_FDT_FILE	"imx6q-b450v3.dtb"
 #elif defined(CONFIG_TARGET_GE_B650V3)
 #define CONFIG_BOARD_NAME	"General Electric B650v3"
-#define CONFIG_DEFAULT_FDT_FILE	"/boot/imx6q-b650v3.dtb"
+#define CONFIG_DEFAULT_FDT_FILE	"imx6q-b650v3.dtb"
 #elif defined(CONFIG_TARGET_GE_B850V3)
 #define CONFIG_BOARD_NAME	"General Electric B850v3"
-#define CONFIG_DEFAULT_FDT_FILE	"/boot/imx6q-b850v3.dtb"
+#define CONFIG_DEFAULT_FDT_FILE	"mx6q-b850v3.dtb"
 #else
 #define CONFIG_BOARD_NAME	"General Electric BA16 Generic"
-#define CONFIG_DEFAULT_FDT_FILE	"/boot/imx6q-ba16.dtb"
+#define CONFIG_DEFAULT_FDT_FILE	"imx6q-ba16.dtb"
 #endif
 
 #define CONFIG_MXC_UART_BASE	UART3_BASE
 #define CONFIG_CONSOLE_DEV	"ttymxc2"
 
+#ifdef CONFIG_YOCTO_IMAGE
+#define CONFIG_BOOT_DIR	""
+#define CONFIG_LOADCMD "fatload"
+#define CONFIG_RFSPART "2"
+#else
+#define CONFIG_BOOT_DIR	"/boot"
+#define CONFIG_LOADCMD "ext2load"
+#define CONFIG_RFSPART "1"
+#endif
+
 #define PHYS_SDRAM_SIZE		(2u * 1024 * 1024 * 1024)
 
 #define CONFIG_SUPPORT_EMMC_BOOT
@@ -144,9 +154,9 @@
 
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	"script=boot.scr\0" \
-	"image=/boot/uImage\0" \
+	"image=" CONFIG_BOOT_DIR "/uImage\0" \
 	"uboot=u-boot.imx\0" \
-	"fdt_file=" CONFIG_DEFAULT_FDT_FILE "\0" \
+	"fdt_file=" CONFIG_BOOT_DIR "/" CONFIG_DEFAULT_FDT_FILE "\0" \
 	"fdt_addr=0x18000000\0" \
 	"boot_fdt=yes\0" \
 	"ip_dyn=yes\0" \
@@ -156,6 +166,8 @@
 	"sddev=0\0" \
 	"emmcdev=1\0" \
 	"partnum=1\0" \
+	"loadcmd=" CONFIG_LOADCMD "\0" \
+	"rfspart=" CONFIG_RFSPART "\0" \
 	"update_sd_firmware=" \
 		"if test ${ip_dyn} = yes; then " \
 			"setenv get_cmd dhcp; " \
@@ -179,12 +191,12 @@
 	"setargs=setenv bootargs console=${console},${baudrate} " \
 		"root=/dev/${rootdev} rw rootwait cma=128M\0" \
 	"loadbootscript=" \
-		"ext2load ${dev} ${devnum}:${partnum} ${loadaddr} ${script};\0" \
+		"${loadcmd} ${dev} ${devnum}:${partnum} ${loadaddr} ${script};\0" \
 	"bootscript=echo Running bootscript from ${dev}:${devnum}:${partnum};" \
 		" source\0" \
 	"loadimage=" \
-		"ext2load ${dev} ${devnum}:${partnum} ${loadaddr} ${image}\0" \
-	"loadfdt=ext2load ${dev} ${devnum}:${partnum} ${fdt_addr} ${fdt_file}\0" \
+		"${loadcmd} ${dev} ${devnum}:${partnum} ${loadaddr} ${image}\0" \
+	"loadfdt=${loadcmd} ${dev} ${devnum}:${partnum} ${fdt_addr} ${fdt_file}\0" \
 	"tryboot=" \
 		"if run loadbootscript; then " \
 			"run bootscript; " \
@@ -237,16 +249,16 @@
 	"usb start; " \
 	"setenv dev usb; " \
 	"setenv devnum 0; " \
-	"setenv rootdev sda1; " \
+	"setenv rootdev sda${rfspart}; " \
 	"run tryboot; " \
 	\
 	"setenv dev mmc; " \
-	"setenv rootdev mmcblk0p1; " \
+	"setenv rootdev mmcblk0p${rfspart}; " \
 	\
 	"setenv devnum ${sddev}; " \
 	"if mmc dev ${devnum}; then " \
 		"run tryboot; " \
-		"setenv rootdev mmcblk1p1; " \
+		"setenv rootdev mmcblk1p${rfspart}; " \
 	"fi; " \
 	\
 	"setenv devnum ${emmcdev}; " \
-- 
2.5.0

